src/hydrusvideodeduplicator/dedup.py

```diff
@@ -370,29 +402,42 @@ def clear_trashed_files_from_db(self) -> None:
 
         try:
             CHUNK_SIZE = 32
-            delete_count = 0
             with SqliteDict(str(DEDUP_DATABASE_FILE), tablename="videos", flag="c") as hashdb:
                 total = len(hashdb)
 
                 if total < 1:
                     return
 
                 rprint(f"[blue] Database found with {total} videos already hashed.")
-                with tqdm(
-                    dynamic_ncols=True,
-                    total=total,
-                    desc="Clearing trashed videos",
-                    unit="video",
-                    colour="BLUE",
-                ) as pbar:
-                    for batched_keys in self.batched(hashdb, CHUNK_SIZE):
-                        is_trashed_result = self.is_files_trashed_hydrus(batched_keys)
-                        for result in is_trashed_result.items():
-                            if result[1] is True:
-                                del hashdb[result[0]]
-                                delete_count += 1
-                        hashdb.commit()
-                        pbar.update(len(batched_keys))
-            self.hydlog.info(f"Cleared {delete_count} trashed videos from the database.")
-        except OSError:
-            rprint("[red] Error while clearing trashed videos cache.")
+                try:
+                    with tqdm(
+                        dynamic_ncols=True,
+                        total=total,
+                        desc="Clearing trashed videos",
+                        unit="video",
+                        colour="BLUE",
+                    ) as pbar:
+                        for batched_keys in self.batched(hashdb, CHUNK_SIZE):
+                            is_trashed_result = self.is_files_trashed_hydrus(batched_keys)
+                            for result in is_trashed_result.items():
+                                if result[1] is True:
+                                    del hashdb[result[0]]
+                            hashdb.commit()
+                            pbar.update(len(batched_keys))
+                except Exception as exc:
+                    rprint("[red] Error while clearing trashed videos cache.")
+                    self.hydlog.info(exc)
+                finally:
+                    delete_count = total - len(hashdb)
+                    if delete_count > 0:
+                        rprint(f"[green] Cleared {delete_count} trashed videos from the database.")
+                        rprint(f"[blue] Updating search indices for remaining files")
+                        self.update_search_cache()
+                    else:
+                        rprint(f"[green] Found no trashed videos to delete from the database.")
+                    rprint(f"[green] There are now {len(hashdb)} videos now the database.")
```

Comment:
@ianwal This all needs to happen within the database context, which is why I added an inner try-except block to the existing try-except block.